tools/sloop.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * Library of functions pertaining to Sloop shadows. Slooping is a way
+ * to simulate the built-in snooping functionality of EotL.
+ *
+ * @author gon
+ */
+
+#define SLOOPEE_SHAD "/usr/gon/tools/sloopee_shad"
+#define SLOOPER_SHAD "/usr/gon/tools/slooper_shad"
+
+// Function Declarations
+protected void add_sloop_shadow( object ob, string str );
+protected void dest_sloop_shadows( object ob, string str );
+protected object * query_sloop_shadows( object ob, string str );
+void add_slooper_shadow( object ob );
+void add_sloopee_shadow( object ob );
+void dest_slooper_shadow( object ob );
+void dest_sloopee_shadow( object ob );
+object find_slooper_shadow( object ob );
+object find_sloopee_shadow( object ob );
+
+/**
+ * Function filters all shdow objects and returns a filtered array of shadows
+ * matching either SLOOPEE_SHAD or SLOOPER_SHAD
+ * @param  object ob   Object passed to check for shadow
+ * @param  string str    string to match (Either SLOOPER_SHAD or SLOOPEE_SHAD)
+ * @return   array of matching shadow(s)
+ */
+protected object * query_sloop_shadows( object ob, string str ) {
+    object * shadow_ob = shadow_list( ob );
+    return filter( shadow_ob, (: return ( program_name( $1 ) == $2 ); :), str );
```

Comment:
This line is more than 80 characters long. For core code, I like to keep things formatted to 80 chars so it still looks good on old school terminals. Modern development environments allow for a lot wider than that, though, so for your own code it's largely up to you what you format to. Some people do 100 chars, some people do 130 or even 160. More importantly, pick a number and stick to it.